sdk/core/azure-core/azure/core/_pipeline_client_async.py

```diff
@@ -43,8 +43,8 @@
 
 _LOGGER = logging.getLogger(__name__)
 
-class _AsyncContextManager(collections.abc.Awaitable):
 
+class _AsyncContextManager(collections.abc.Awaitable):
```

Comment:
https://docs.python.org/3/library/contextlib.html#contextlib.AbstractAsyncContextManager

Reply:
The code is a little more complicated than that, I think the name `_AsyncContextManager` is misleading, or maybe it was changed over time and became more than that. To be investigated.

Reply:
@johanste what you had in mind? https://github.com/Azure/azure-sdk-for-python/issues/26760

Reply:
Ok, this one was tricky, I spent like 2 hours on it today, I renamed a few things for clarity, I think the code is cleaner now. Let me know your thoughts @johanste